app/src/main/java/me/niccorder/news/list/NewsListActivity.java

```diff
@@ -91,4 +117,83 @@ private void initRecycler() {
         newsAdapter = new NewsAdapter(articles);
         newsRecycler.setAdapter(newsAdapter);
     }
+
+    /**
+     * To be called when a PTR even has occurred.
+     */
+    private void onRefresh() {
+        Log.d(TAG, "onRefresh()");
+
+        newsRepository.getArticles(new RepositoryCallback<List<Article>>() {
+            @Override
+            public void onSuccess(List<Article> data) {
+                super.onSuccess(data);
+
+                updateArticles(articles, data);
+                refreshLayout.setRefreshing(false);
+            }
+
+            @Override
+            public void onError(Throwable error) {
+                super.onError(error);
+
+                refreshLayout.setRefreshing(false);
+            }
+        });
+    }
+
+    /**
+     * Updates the articles given the old and new datasets.
+     *
+     * @param oldArticles to be updated.
+     * @param newArticles to be displayed.
+     */
+    private void updateArticles(
+            @NonNull final List<Article> oldArticles,
+            @NonNull final List<Article> newArticles
+    ) {
+        Log.d(TAG, "updateArticles()");
+
+        // DiffUtil is a utility provided by the android support libraries to calculate the
+        // diff between two lists for us. Thankfully, it also works seamlessly with a recyclerview
+        // adapter as you will see below.
+        final DiffUtil.DiffResult diffResult = DiffUtil.calculateDiff(new DiffUtil.Callback() {
```

Comment:
This should be fine for the MVP, but we will need to create a task to calculate the Diff on the background thread. If we calculate the diff on the main thread, it will definitely lock up the UI for larger lists.

Looks good for this PR though!